### ISSUE TITLE
negation of SC2056, `(( $1 == 0 && $1 == 3 ))`, is not detected

### ISSUE THREAD

**Issue by e-kwsm** [koalaman/shellcheck#3181]
#### For bugs
- Rule Id (if any, e.g. SC1000): [SC2056](https://www.shellcheck.net/wiki/SC2056)?
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
# ! (( $1 != 0 || $1 != 3 ))  # SC2056 is detected
if (( $1 == 0 && $1 == 3 ))
then
  echo "$1 is 0 or 3"
fi
```

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

```
Line 3:
if (( $1 == 0 && $1 == 3 ))
              ^-- SC2056 (warning): You probably wanted || here, otherwise it's always false.
```
